mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -70,12 +70,13 @@ private[clustering] trait GaussianMixtureParams extends Params with HasMaxIter w
 @Experimental
 class GaussianMixtureModel private[ml] (
     @Since("2.0.0") override val uid: String,
-    private val parentModel: MLlibGMModel)
+    private val parentModel: MLlibGMModel,
+    @Since("2.0.0") @transient protected val sqlContext: SQLContext)
```

Comment:
Sorry, but I'd like to change my recommendation.  To follow existing implementations, let's just getOrCreate the SQLContext in gaussiansDF.